make.py

```diff
@@ -0,0 +1,249 @@
+    A library will match if the pattern is a substring of the library's name.
+
+    :type from_dir: str
+    :type destination: str
+    :type patterns: list[str]
+    """
+    for root, _, filenames in os.walk(from_dir):
+        for filename in filenames:
+            if any(n in filename for n in patterns):
+                if any(ext in filename for ext in (".so", ".dll", ".dylib")):
+                    shutil.copy(os.path.join(root, filename), destination)
+
+
+def manage(lang_dir, verb, build_mode, *add_args):
+    """
+    Run the manage.py script from of a Langkit-based language definition.
+
+    Raises CalledProcessError if 'manage.py' terminates with non-zero status.
+
+    :param str verb: 'generate', 'build', 'install', etc.
+    :param str build_mode: 'dev' or 'prod'.
+    :param list[str] add_args: additional arguments to pass
+    """
+    with in_dir(lang_dir):
+        args = [sys.executable, 'manage.py', verb]
+        args.extend(['--build-mode', build_mode])
+        args.extend(add_args)
+
+        if verb in ('build', 'make'):
+            # Always build shared libs with "$ORIGIN" rpaths.
+            args.append('--gargs=-R')
+            env = os.environ.copy()
+            env['LD_RUN_PATH'] = "$ORIGIN"
+        else:
+            env = os.environ
+
+        subprocess.check_call(args, env=env)
+
+
+def build_language(lang_dir, build_mode):
+    """
+    Build a Langkit-based language.
+
+    Raises CalledProcessError if build fails.
+
+    :param str lang_dir: The directory in which the 'manage.py' script lives.
+    :param str build_mode: The build mode to use (dev or prod).
+    """
+    manage(lang_dir, 'make', build_mode, '-P')
+
+
+def install_language(lang_dir, lang_name, build_mode, prefix,
+                     gnat_install, gnatcoll_bindings_install, libgpr_install):
+    """
+    Install a Langkit-based language and its python bindings.
+
+    Raises CalledProcessError or OSError if build fails.
+
+    :param str lang_dir: The directory in which the 'manage.py' script lives.
+    :param str lang_name: The full name of the language
+        (e.g. libadalang, libpythonlang).
+    :param str build_mode: The build mode to use (dev or prod).
+    :param str prefix: Where to install the language.
+
+    :param str gnat_install: The directory in which GNAT is installed.
+    :param str gnatcoll_bindings_install: The directory in which GNATCOLL
+        bindings are installed.
+    :param str libgpr_install: The directory in which libgpr is installed.
+    """
+    manage(lang_dir, 'install', build_mode, prefix)
+
+    # Copy all necessary shared libraries in the python package, so that the
+    # subsequent call to pip_install packages all of them.
+
+    lang_python_dir = os.path.join(lang_dir, 'build', 'python')
+    dest = os.path.join(lang_python_dir, lang_name)
+
+    copy_all_shared_libs(gnat_install, dest, "gnat", "gnarl", "xmlada")
+    copy_all_shared_libs(gnatcoll_bindings_install, dest, "gnatcoll")
```

Comment:
This is not complete: both xmlada and gnatcoll-core can be installed outside of gnat. I suggest you import the whole logic from `libadalang_standalone.anod`: not only we’ll be sure that it will work on all supported platforms, but this will also enable us to remove that logic from there (and keep it in only one place).